wellcomeml/ml/voting_classifier.py

```diff
@@ -49,23 +77,22 @@ def predict(self, X):
                     return np.argmax(Y_prob, axis=1)
             else:  # hard voting
 
+                # If num_agree isn't set then use majority vote
+                if not self.num_agree:
+                    # So if 4 estimators, >= 3 need to agree
+                    self.num_agree = np.ceil((len(estimators) + 1) / 2)
+
                 Y_preds = [est.predict(X) for est in estimators]
                 Y_preds = np.array(Y_preds)
                 if self.multilabel:
-                    if self.num_agree:
-                        logger.warning("num_agree specified but not needed for multilabel voting")
-                    return np.apply_along_axis(
-                        lambda x: np.argmax(np.bincount(x)), axis=0, arr=Y_preds
-                    )
+                    return np.array(Y_preds.sum(axis=0) >= self.num_agree, dtype='int32')
                 else:
-                    # If num_agree isn't set then use majority vote
-                    if not self.num_agree:
-                        # So if 4 estimators, >= 3 need to agree
-                        self.num_agree = np.ceil((len(estimators) + 1) / 2)
-
-                    return np.apply_along_axis(
-                        lambda x: int(np.sum(x) >= self.num_agree), axis=0, arr=Y_preds
-                    )
+                    votes = np.apply_along_axis(lambda x: max(np.bincount(x)), axis=0, arr=Y_preds)
+                    max_class = np.apply_along_axis(lambda x: np.argmax(np.bincount(x)), axis=0, arr=Y_preds)
+                    # If no maximum over the threshold, then pick the first from an ordered list
+                    # of the other options, e.g. if 5,2,3 were voted on pick 2 (not 0)
+                    options = np.sort(np.transpose(Y_preds))
```

Comment:
options might be a bit confusing as a name. also it took me some time to get my head around what Y_preds contains at this point and why you need to transpose, maybe a small comment to say about the dimensions of Y_preds at this point e.g. Y_preds: n_estimators x n_examples with predicted class as values